great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -138,6 +158,84 @@ def _validate_batch_request(self, batch_request: BatchRequest) -> None:
+    ) -> List[Batch]:
+        self._validate_batch_request(batch_request)
+
+        execution_engine: PandasExecutionEngine | SparkDFExecutionEngine = (
+            self.datasource.get_execution_engine()
+        )
+
+        data_connector: DataConnector = self._get_data_connector()
+
+        batch_definition_list: List[
+            BatchDefinition
+        ] = data_connector.get_batch_definition_list_from_batch_request(
+            batch_request=batch_request
+        )
+
+        batch_list: List[Batch] = []
+
+        batch_spec: BatchSpec
+        batch_spec_options: dict
+        batch_data: Any
+        batch_markers: BatchMarkers
+        batch_metadata: BatchRequestOptions
+        batch: Batch
+        for batch_definition in batch_definition_list:
+            batch_spec = data_connector.build_batch_spec(
+                batch_definition=batch_definition
+            )
+            batch_spec_options = {
+                "reader_method": self._get_reader_method(),
+                "reader_options": self.dict(
+                    include=self._get_reader_options_include(),
+                    exclude=self._EXCLUDE_FROM_READER_OPTIONS,
+                    exclude_unset=True,
+                    by_alias=True,
+                ),
+            }
+            batch_spec.update(batch_spec_options)
+
+            batch_data, batch_markers = execution_engine.get_batch_data_and_markers(
+                batch_spec=batch_spec
+            )
+
+            batch_metadata = copy.deepcopy(batch_request.options)
+            # TODO: <Alex>ALEX-FIX_TO_INSURE_BASE_DIRECTORY_TYPE_IS_PATHLIB.PATH-CONSISTENTLY</Alex>
+            # TODO: <Alex>ALEX</Alex>
+            # TODO: <Alex>ALEX-FIX_TO_INSURE_PROPERTY_NAME_CORRESPONDDS_TO_ITS_MEANING</Alex>
+            batch_metadata["base_directory"] = pathlib.Path(batch_spec["path"])
+            # TODO: <Alex>ALEX</Alex>
+            # TODO: <Alex>ALEX</Alex>
+            batch_metadata.update(batch_definition.batch_identifiers)
+            batch_request.options.update(batch_definition.batch_identifiers)
+            # TODO: <Alex>ALEX</Alex>
+
+            # Some pydantic annotations are postponed due to circular imports.
+            # Batch.update_forward_refs() will set the annotations before we
+            # instantiate the Batch class since we can import them in this scope.
+            Batch.update_forward_refs()
+
+            batch = Batch(
+                datasource=self.datasource,
+                data_asset=self,
+                batch_request=batch_request,
+                data=batch_data,
+                metadata=batch_metadata,
+                legacy_batch_markers=batch_markers,
+                legacy_batch_spec=batch_spec,
+                legacy_batch_definition=batch_definition,
+            )
+            batch_list.append(batch)
+
+        # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
```

Comment:
what's the level of effort in resolving these TODOs around sorting? They seem pretty prevalent.

Reply:
@donaldheppner These are "notes to self" -- the level-of-effort should be 3 -- 4 hours (calendar time, including the provisions for a standard working schedule having regular meetings, etc.).  However, I feel that this should be part of the very next pull-request for the reasons of the principle of "committing smallest amount of code that improves GX".  Thanks.